온보딩 화면 리팩토링

theBettor opened the pull request:
## 🤷‍♂️ PR 내용(Issue 꼭 달기!)
- 여러분들의 도움으로 인해 그나마 유의미한 PR을 올리게 되었습니다. 감사합니다!
- 주요 내용은 아래와 같습니다. 
resolved: #49, #80, #82, #94 #95 
  1. 변경된 피그마대로 디자인 화면 리펙토링(기획 단계에서 변경 안된 부분이 있음).
  2. 흐름에 맞는 화면 연결

- 향후 추가될 내용
  - 기획 단계에서 정리가 안된 '그룹 만들기' 부분의 작업이 명확해짐에 따라 ViewModel 부분 수정이 필요할 것(registerFamily가 쓰여지지 않고 있음).

- 질문
  - Q1. 지금 구현되어있는 로직이 '로그인만' 되었을때, 홈 화면 보여주게 되어있는지, 그렇다면 로그인은 되었는데 권한 설정이 안되있거나, 가족 공간이 없을 때에는 어떻게 처리해야하는지?
  - Q2. OnBoardingLayout을 사용하다보니 첨부한 사진의 맨 마지막 화면처럼 프로필 이미지의 위치를 조정할 수 없는데 어떻게 하면 좋을지?
![스크린샷 2025-01-20 오후 11 11 40](https://github.com/user-attachments/assets/09ac0397-a80f-48cb-b975-be94c25e981c)

theBettor committed:
Merge branch 'develop' into refactor/onboarding

theBettor committed:
온보딩 화면으로 가기

theBettor committed:
refactor: BasicTextField 이용하여 초대코드 입력 받는 부분 수정

theBettor committed:
refactor: 공간 입장하기시, 홈화면으로 가는 로직 추가

theBettor committed:
question: 아래 2개의 route는 왜 사용되지 않는건지 궁금하다.

theBettor committed:
style: 주석 변경, 코드 포맷팅 수정

theBettor committed:
refactor: 에러메시지 핸들링 관련 코드 삭제, 주석 정리

agvber commented:
Q1. 지금 구현되어있는 로직이 '로그인만' 되었을때, 홈 화면 보여주게 되어있는지, 그렇다면 로그인은 되었는데 권한 설정이 안되있거나, 가족 공간이 없을 때에는 어떻게 처리해야하는지?
> 이건 기획적인 부분이라 수빈님이랑 같이 이야기 하면 좋을 것 같네요

Q2. OnBoardingLayout을 사용하다보니 첨부한 사진의 맨 마지막 화면처럼 프로필 이미지의 위치를 조정할 수 없는데 어떻게 하면 좋을지?
> 프로필 위치를 padding으로 주면 될 것 같은데 정확히 이해가 되지 않네요... (아니면 OnboardingLayout을 사용하지 않거나...?)

theBettor committed:
feat: post notification 분기 처리

DJDrama commented on feature/onboarding/src/main/java/com/teampatch/feature/onboarding/login/ui/OnboardingEnterInvitationCodeScreen.kt:
```diff
@@ -148,25 +194,32 @@ fun InputProfileSettings() {
 
-            Image(painter = painterResource(com.teampatch.core.designsystem.R.drawable.btn_enter_space), null)
+            Image(
+                painter = painterResource(com.teampatch.core.designsystem.R.drawable.btn_enter_space),
```
com.teampatch.core.designsystem.

패키지 정보 import로 없애버리세요~!

theBettor replied:
이걸 빠뜨렸네요 감사합니다

DJDrama commented on feature/onboarding/src/main/java/com/teampatch/feature/onboarding/login/ui/OnboardingFirstScreen.kt:
```diff
@@ -40,7 +40,6 @@ internal fun OnboardingFirstScreen(
-    val errorMessage by viewModel.errorMessage.collectAsState()
     val context = LocalContext.current
 
     LaunchedEffect(isLoginSuccessful) {
```
if(isLoginSuccessful) 이 추가되어야 할 것 같습니다~!
isLoginSuccessful이 true에서 false로 바뀌어도 실행될텐데 괜찮은 부분일까요??

theBettor replied:
부끄럽지만 성공/실패에 대한 처리에 대한 아이디어가 지금 당장은 떠오르지가 않아요.. 그래서 fasle때 처리해주는 로직도 현재는 없어서 고민 해봐야될 것 같아요 @DJDrama

theBettor replied:
https://github.com/Team-PATCH/Harmony-Android/pull/89/commits/536751f25f1cd63592ccfd96faac485e4c66fabc @DJDrama 해결했습니다.

theBettor committed:
WIP : 왜 뒤로가기 버튼이 동작하지 않을까

theBettor committed:
style : trailing comma 누락

theBettor committed:
refactor: painterResource import로 간소화

theBettor commented:
> Q1. 지금 구현되어있는 로직이 '로그인만' 되었을때, 홈 화면 보여주게 되어있는지, 그렇다면 로그인은 되었는데 권한 설정이 안되있거나, 가족 공간이 없을 때에는 어떻게 처리해야하는지?
> 
> > 이건 기획적인 부분이라 수빈님이랑 같이 이야기 하면 좋을 것 같네요
> 
> Q2. OnBoardingLayout을 사용하다보니 첨부한 사진의 맨 마지막 화면처럼 프로필 이미지의 위치를 조정할 수 없는데 어떻게 하면 좋을지?
> 
> > 프로필 위치를 padding으로 주면 될 것 같은데 정확히 이해가 되지 않네요... (아니면 OnboardingLayout을 사용하지 않거나...?)

theBettor committed:
wip: 백버튼

theBettor committed:
Merge branch 'develop' into refactor/onboarding

theBettor committed:
style : apply ktlint formatting

theBettor committed:
refactor: 로그인 이벤트 흐름을, 상태를 collect 하지말고 sharedFlow, Channel을 사용해서 event를 collect

theBettor committed:
refactor: 로그인 이벤트 흐름을, 상태를 collect 하지말고 sharedFlow, Channel을 사용해서 event를 collect #94

theBettor committed:
fix: 로그인 실패시의 분기처리

theBettor committed:
fix: 로그인 실패시의 분기처리

theBettor commented:
https://github.com/Team-PATCH/Harmony-Android/pull/89/commits/8f0a0da548ec36826173b490cd76bd6a6ffadd5c @agvber 해결했습니다

agvber commented on feature/onboarding/src/main/java/com/teampatch/feature/onboarding/login/ui/OnboardingEnterInvitationCodeScreen.kt:
```diff
@@ -18,34 +27,38 @@ import androidx.compose.runtime.remember
-    // 버튼 활성화 여부 (5자리 모두 입력되면 활성화)
-    val isNextEnabled = code.length == 5
+    var code by remember { mutableStateOf(listOf("", "", "", "", "")) }
+    val isNextEnabled = code.all { it.isNotEmpty() } // 5자리가 모두 입력되면 활성화
```
해당부분 코드는 recomposition 될때마다 재실행되는 문제가 있어 성능 저하가 일어날 수 있어요
remember로 감싸서 상태 관리를 하는 것은 어떨까요??

https://youtu.be/EOQB8PTLkpY?si=BzqhKAfRTpUds84S&t=114

theBettor replied:
https://github.com/Team-PATCH/Harmony-Android/issues/95

agvber commented on feature/onboarding/src/main/java/com/teampatch/feature/onboarding/login/ui/OnboardingFirstScreen.kt:
```diff
@@ -39,17 +39,29 @@ internal fun OnboardingFirstScreen(
+    LaunchedEffect(Unit) {
+        viewModel.loginSuccessEvent.collect { isLoginSuccessful ->
+            if (isLoginSuccessful) {
+                val hasNotificationGranted =
```
해당부분은 함수로 뺴주면 좋을 것 같아요!

fun hasNotificationGranted() {
}

theBettor replied:
#96

theBettor committed:
perf : 불필요한 recomposition 방지 #95

객체 자체는 유지하면서 내부 요소만 변경 가능하게 해야하며, 요소가 변하지 않으면 불필요한 계산을 하지 않아야하는 등 불필요한 recomposition이 일어나지 않게 해야함.

theBettor committed:
perf : 불필요한 recomposition 방지 #95

객체 자체는 유지하면서 내부 요소만 변경 가능하게 해야하며, 요소가 변하지 않으면 불필요한 계산을 하지 않아야하는 등 불필요한 recomposition이 일어나지 않게 해야함.

theBettor committed:
perf: hasNotificationGranted를 함수로 분리 #96

theBettor committed:
perf: hasNotificationGranted를 함수로 분리 #96

theBettor committed:
Merge remote-tracking branch 'origin/refactor/onboarding' into refactor/onboarding
